Blizzard.eai

```diff
@@ -587,24 +587,19 @@ function ZoomUp takes nothing returns nothing
 endfunction
 
 function InitZoom takes nothing returns nothing
-  local integer i = 0
-  if ZoomSetx == null then
-    set ZoomSetx = CreateTrigger()
-    set AdjustZoomUp = CreateTrigger()
-    set AdjustZoomDown = CreateTrigger()
-    call TriggerAddAction(ZoomSetx, function ZoomSet)
-    //call TriggerAddAction(AdjustZoomUp , function ZoomUp)
-    //call TriggerAddAction(AdjustZoomDown , function ZoomDown)
-    //loop
-    //  exitwhen i >= GetBJMaxPlayers()
-      //call SetCameraFieldForPlayer( Player(i), CAMERA_FIELD_TARGET_DISTANCE, 2500, 0 ) // not required in latest war3
-      if (GetPlayerController(GetLocalPlayer()) != MAP_CONTROL_COMPUTER) or IsPlayerObserver(GetLocalPlayer()) then
-        call TriggerRegisterPlayerChatEvent( ZoomSetx, GetLocalPlayer(), "-zoom", false )
-        //call TriggerRegisterPlayerKeyEventBJ( AdjustZoomUp, Player(i), bj_KEYEVENTTYPE_DEPRESS, bj_KEYEVENTKEY_UP ) // not required in latest war3
-        //call TriggerRegisterPlayerKeyEventBJ( AdjustZoomDown, Player(i), bj_KEYEVENTTYPE_DEPRESS, bj_KEYEVENTKEY_DOWN )
-      endif
-    //  set i = i + 1
-    //endloop
+  set ZoomSetx = CreateTrigger()
+  set AdjustZoomUp = CreateTrigger()
+  set AdjustZoomDown = CreateTrigger()
+  call TriggerAddAction(ZoomSetx, function ZoomSet)
+  call TriggerAddAction(AdjustZoomUp , function ZoomUp)
+  call TriggerAddAction(AdjustZoomDown , function ZoomDown)
+  if (GetPlayerController(GetLocalPlayer()) != MAP_CONTROL_COMPUTER) and not IsPlayerObserver(GetLocalPlayer()) then
+    call DisplayTimedTextToPlayer(GetLocalPlayer(),0,0,7,"Set Zoom type: |c00d5f038-zoom1850|r")
+    call TriggerRegisterPlayerChatEvent( ZoomSetx, GetLocalPlayer(), "-zoom", false )
+    call TriggerRegisterPlayerKeyEventBJ( AdjustZoomUp, GetLocalPlayer(), bj_KEYEVENTTYPE_DEPRESS, bj_KEYEVENTKEY_UP ) // not required in latest war3
+    call TriggerRegisterPlayerKeyEventBJ( AdjustZoomDown, GetLocalPlayer(), bj_KEYEVENTTYPE_DEPRESS, bj_KEYEVENTKEY_DOWN )
+  elseif IsPlayerObserver(GetLocalPlayer()) then
+    call SetCameraFieldForPlayer(GetTriggerPlayer(), CAMERA_FIELD_TARGET_DISTANCE, 1950, 0)
```

Comment:
Similarly conflicts with official support by blizzard so I won't include default distance set as players can configure defaults themselves which may not match this number.